# Analyse des jeux de données data_sig_collecte

Il y a 2 jeux de données :
* si : societes immatriculees (2020)
* villes_data : données sur les entreprises au niveau national

## Importation des librairies

In [44]:
import pandas as pd

## Lecture des fichiers

In [50]:
si = pd.read_csv("../Data/societes-immatriculees-2020.csv", index_col= 0, sep=';', skiprows=0)
data_villes = pd.read_excel("/home/simplon/Documents/Cliink_Equipe2/Data/villes_data.xlsx", index_col= 0, sep=';', skiprows=0)
entreprises_par_sa = pd.read_csv("../Data/villes_data.csv", header=1 ,encoding="UTF-8", sep=',', skiprows=list(range(0, 2)) + list(range(14, 245)), usecols=[i for i in range(12)])
#entreprises_par_sa_ef = pd.read_csv("../Data/villes_data.csv", header=16 ,encoding="UTF-8", sep=',', skiprows=list(range(15, 19)) + list(range(28, 245)), usecols=[i for i in range(31)])
#année_2017 = pd.read_csv("../Data/villes_data.csv", header=30 ,encoding="UTF-8", sep=',', skiprows=list(range(29, 34)) + list(range(37, 245)), usecols=[i for i in range(14)])
#pop_age = pd.read_csv("../Data/villes_data.csv", header=39 ,encoding="UTF-8", sep=',', skiprows=list(range(38, 41)) + list(range(50, 245)), usecols=[i for i in range(13)])
#CSP_ParAge = pd.read_csv("../Data/villes_data.csv", header=52 ,encoding="UTF-8", sep=',', skiprows=list(range(51, 55)) + list(range(66, 245)), usecols=[i for i in range(6)])
#CSP_ParAgeSexe = pd.read_csv("../Data/villes_data.csv", header=68 ,encoding="UTF-8", sep=',', skiprows=list(range(67, 71)) + list(range(82, 245)), usecols=[i for i in range(11)])
#CSP_15_24 = pd.read_csv("../Data/villes_data.csv", header=84 ,encoding="UTF-8", sep=',', skiprows=list(range(83, 88)) + list(range(99, 245)), usecols=[i for i in range(6)])
#CSP_25_54 = pd.read_csv("../Data/villes_data.csv", header=101 ,encoding="UTF-8", sep=',', skiprows=list(range(100, 105)) + list(range(116, 245)), usecols=[i for i in range(6)])
#CSP_55_plus = pd.read_csv("../Data/villes_data.csv", header=118 ,encoding="UTF-8", sep=',', skiprows=list(range(117, 122)) + list(range(133, 245)), usecols=[i for i in range(6)])
#emploi_age = pd.read_csv("../Data/villes_data.csv", header=135 ,encoding="UTF-8", sep=',', skiprows=list(range(134, 138)) + list(range(144, 245)), usecols=[i for i in range(16)])
#emploi_femme_age = pd.read_csv("../Data/villes_data.csv", header=145 ,encoding="UTF-8", sep=',', skiprows=list(range(145, 150)) + list(range(161, 245)), usecols=[i for i in range(7)])
#emploi_homme_age = pd.read_csv("../Data/villes_data.csv", header=162 ,encoding="UTF-8", sep=',', skiprows=list(range(162, 167)) + list(range(178, 245)), usecols=[i for i in range(7)])
#chomage_age = pd.read_csv("../Data/villes_data.csv", header=179 ,encoding="UTF-8", sep=',', skiprows=list(range(179, 184)) + list(range(190, 245)), usecols=[i for i in range(7)])
#taux_chomage_age_sexe = pd.read_csv("../Data/villes_data.csv", header=192 ,encoding="UTF-8", sep=',', skiprows=list(range(191, 196)) + list(range(202, 245)), usecols=[i for i in range(13)])
#emploi_catégorie_socio_pro = pd.read_csv("../Data/villes_data.csv", header=204 ,encoding="UTF-8", sep=',', skiprows=list(range(203, 209)) + list(range(218, 245)), usecols=[i for i in range(7)])
#logement_par_catégorie = pd.read_csv("../Data/villes_data.csv", header=220 ,encoding="UTF-8", sep=',', skiprows=list(range(219, 224)) + list(range(230, 245)), usecols=[i for i in range(7)])


In [56]:
sa_établissementActifs

Catégories  Ensemble  0 salarié  \
0                                           Ensemble      3416       2520   
1                 Agriculture. sylviculture et pêche        17         14   
2                                          Industrie       161        107   
3                                       Construction       433        335   
4               Commerce. transport. services divers      2436       1763   
5           - dont commerce et réparation automobile       614        371   
6  Administration publique. enseignement. santé. ...       369        301   

   1 à 9 salariés  10 à 19 salariés  20 à 49 salariés  50 salariés ou plus  \
0             760                80                39                   17   
1               2                 1                 0                    0   
2              41                 9                 3                    1   
3              86                11                 1                    0   
4             582                51                29                   11   
5             203                19                14                    7   
6              49                 8                 6                    5   

   Ensemble.1  0 salarié.1  1 à 9 salariés.1  ...  1 à 9 salariés.3  \
0        3707         2675               831  ...              3494   
1          22           20                 2  ...                 6   
2         185          114                51  ...               103   
3         511          343               142  ...               206   
4        2574         1886               573  ...              2939   
5         563          373               155  ...               981   
6         415          312                63  ...               240   

   10 à 19 salariés.3  20 à 49 salariés.3  50 salariés ou plus.3  Ensemble.4  \
0                 286                 163                     85       62220   
1                   0                   1                      0         345   
2                  13                   7                      1        2703   
3                  17                   2                      0        4616   
4                 211                 105                     45       44753   
5                  46                  29                      9        9363   
6                  45                  48                     39        9803   

   0 salarié.4  1 à 9 salariés.4  10 à 19 salariés.4  20 à 49 salariés.4  \
0        44061             13299                2219                1485   
1          236                98                   4                   5   
2         1576               682                 204                 124   
3         3398               892                 140                 122   
4        31354             10461                1449                 890   
5         5640              2936                 415                 230   
6         7497              1166                 422                 344   

   50 salariés ou plus.4  
0                   1156  
1                      2  
2                    117  
3                     64  
4                    599  
5                    142  
6                    374  

[7 rows x 31 columns]

In [55]:
si = pd.read_csv("../Data/societes-immatriculees-2020.csv", index_col= 0, sep=';', skiprows=0)
data_villes = pd.read_csv("../Data/villes_data.csv", index_col= 0, sep=';', skiprows=0)
#sa_nomSecteurActivité = pd.read_csv("../Data/villes_data.csv", header=1 ,encoding="UTF-8", sep=',', skiprows=list(range(0, 2)) + list(range(14, 245)), usecols=[i for i in range(12)])
sa_établissementActifs = pd.read_csv("../Data/villes_data.csv", header=16 ,encoding="UTF-8", sep=',', skiprows=list(range(15, 19)) + list(range(28, 245)), usecols=[i for i in range(31)])
#ha_année_2017 = pd.read_csv("../Data/villes_data.csv", header=30 ,encoding="UTF-8", sep=',', skiprows=list(range(29, 34)) + list(range(37, 245)), usecols=[i for i in range(14)])
#ha_age = pd.read_csv("../Data/villes_data.csv", header=39 ,encoding="UTF-8", sep=',', skiprows=list(range(38, 41)) + list(range(50, 245)), usecols=[i for i in range(13)])
#ha_emploiParAge = pd.read_csv("../Data/villes_data.csv", header=52 ,encoding="UTF-8", sep=',', skiprows=list(range(51, 55)) + list(range(66, 245)), usecols=[i for i in range(6)])



CSP_ParAgeSexe = pd.read_csv("../Data/villes_data.csv", header=68 ,encoding="UTF-8", sep=',', skiprows=list(range(67, 71)) + list(range(82, 245)), usecols=[i for i in range(11)])
CSP_15_24 = pd.read_csv("../Data/villes_data.csv", header=84 ,encoding="UTF-8", sep=',', skiprows=list(range(83, 88)) + list(range(99, 245)), usecols=[i for i in range(6)])
CSP_25_54 = pd.read_csv("../Data/villes_data.csv", header=101 ,encoding="UTF-8", sep=',', skiprows=list(range(100, 105)) + list(range(116, 245)), usecols=[i for i in range(6)])
CSP_55_plus = pd.read_csv("../Data/villes_data.csv", header=118 ,encoding="UTF-8", sep=',', skiprows=list(range(117, 122)) + list(range(133, 245)), usecols=[i for i in range(6)])

emploi_femme = pd.read_csv("../Data/villes_data.csv", header=145 ,encoding="UTF-8", sep=',', skiprows=list(range(145, 150)) + list(range(161, 245)), usecols=[i for i in range(7)])
emploi_homme = pd.read_csv("../Data/villes_data.csv", header=162 ,encoding="UTF-8", sep=',', skiprows=list(range(162, 167)) + list(range(178, 245)), usecols=[i for i in range(7)])
chomage_sexe = pd.read_csv("../Data/villes_data.csv", header=179 ,encoding="UTF-8", sep=',', skiprows=list(range(179, 184)) + list(range(190, 245)), usecols=[i for i in range(7)])
taux_chomage_age = pd.read_csv("../Data/villes_data.csv", header=192 ,encoding="UTF-8", sep=',', skiprows=list(range(191, 196)) + list(range(202, 245)), usecols=[i for i in range(13)])
logement_par_catégorie = pd.read_csv("../Data/villes_data.csv", header=220 ,encoding="UTF-8", sep=',', skiprows=list(range(219, 224)) + list(range(230, 245)), usecols=[i for i in range(7)])

In [57]:
ha_CSP_tranche_age = [CSP_ParAgeSexe, CSP_15_24, CSP_25_54, CSP_55_plus]
ha_CSP_tranche_age

AttributeError: 'list' object has no attribute 'head'

In [52]:
sa_nomSecteurActivité

Unnamed: 0   France  Mougins  \
0  Industrie manufacturière. industries extractiv...   393285      210   
1                                       Construction   685872      565   
2  Commerce de gros et de détail. transports. héb...  1673387      978   
3                       Information et communication   210974      147   
4               Activités financières et d'assurance   231445      184   
5                             Activités immobilières   262687      362   
6  Activités spécialisées. scientifiques et techn...  1031096      807   
7  Administration publique. enseignement. santé h...   745198      488   
8                       Autres activités de services   436388      280   
9                                              Total  5670332     4021   

   Mandelieu-la-Napoule  Le Cannet  Théoule-sur-Mer  Cannes  France.1  \
0                   180        211                7     477       6.9   
1                   422        816               28    1344      12.1   
2                   998       1160               95    4815      29.5   
3                    72        136                6     351       3.7   
4                   133        172               10     618       4.1   
5                   268        368               57    1705       4.6   
6                   654       1033               55    2588      18.2   
7                   385        679               24    1482      13.1   
8                   311        426               31    1158       7.7   
9                  3423       5001              313   14538     100.0   

   Mougins.1  Mandelieu-la-Napoule.1  Le Cannet.1  Théoule-sur-Mer.1  
0        5.2                     5.3          4.2                2.2  
1       14.1                    12.3         16.3                8.9  
2       24.3                    29.2         23.2               30.4  
3        3.7                     2.1          2.7                1.9  
4        4.6                     3.9          3.4                3.2  
5        9.0                     7.8          7.4               18.2  
6       20.1                    19.1         20.7               17.6  
7       12.1                    11.2         13.6                7.7  
8        7.0                     9.1          8.5                9.9  
9      100.0                   100.0        100.0              100.0

## Jeu de données des entreprises et de la population
### Aperçu du fichier

Il contient 26 colonnes que l'on pourra étudier :


#### Pour les entreprises :

* Dénomination
* Siren
* Secteur d'activité
* Adresse
* Code postal
* Ville
* Géolocalisation

#### Par secteur d'activité : 

* Secteur d’activité
* Emplois selon le secteur d’activité
* Nombre d'établissements au 31 décembre 2018 par secteur d'activité
* Établissements actifs par secteur d'activité et tranche d'effectif salarié au 31 décembre

#### Par ville : 

* Population de 2017
* Population par grandes tranches d'âge
* Population de 15 ans ou plus par CSP
* Population de 15 ans ou plus par sexe et CSP
* Population des 15-24 ans par CSP
* Population des 25-54 ans par CSP
* Population des 55 ans ou plus par CSP
* Activité et emploi de la population de 15 à 64 ans par âge
* Statut et condition d'emploi des FEMMES de 15 ans ou plus
* Statut et condition d'emploi des HOMMES de 15 ans ou plus
* Chômage (au sens du recensement) des 15-64 ans
* Taux de chômage (au sens du recensement) par sexe et âge
* Emplois par catégorie socioprofessionnelle
* Logements par catégorie
* Logements selon leurs types


Parmis lesquelles les variables suivantes ont été extraites :

* forme_juridique : fromes juridiques des entreprises
* si_cannes : nombre d'entreprises à Cannes
* cinq_villes : variable regroupant l'ensemble des données des entreprises présentes dans les cinq villes de l'agglomération
* fj : forme juridique

In [ ]:
'''
Par secteur d'activité :

* Secteur d’activité
* entreprises_par_sa : Nombre d'établissements au 31 décembre 2018 par secteur d'activité
* entreprises_par_sa_ef : Établissements actifs par secteur d'activité et tranche d'effectif salarié au 31 décembre

Par ville : 

* pop_2017 : Population de 2017
* pop_age : Population par grandes tranches d'âge
* CSP_ParAge : Population de 15 ans ou plus par CSP
* CSP_ParAgeSexe : Population de 15 ans ou plus par sexe et CSP
* CSP_15_24 : Population des 15-24 ans par CSP
* CSP_25_54 : Population des 25-54 ans par CSP
* CSP_55_plus : Population des 55 ans ou plus par CSP
* emploi_age : Activité et emploi de la population de 15 à 64 ans par âge
* emploi_femme_age : Statut et condition d'emploi des FEMMES de 15 ans ou plus
* emploi_homme_age : Statut et condition d'emploi des HOMMES de 15 ans ou plus
* chomage_age : Chômage (au sens du recensement) des 15-64 ans
* taux_chomage_age_sexe : Taux de chômage (au sens du recensement) par sexe et âge
* emploi_catégorie_socio_pro : Emplois par catégorie socioprofessionnelle
* logement_par_catégorie : Logements par catégorie
* logement_par_types : Logements selon leurs types

'''

# Faire ressortir uniquement les données des cinq villes de l'agglomération du jeu de données 

### Décompte du nombre d'entreprises présentes dans le jeu de données par villes qui nous interesse

* Cannes : 368 entreprises
* Le Cannet : 119 entreprises
* Teoule : 9 entreprises
* Mandelieu : 84 entreprises
* Mougins : 124 entreprises

Nombre total d'entreprises dans l'agglomération : 704

### Exemple avec la ville de Cannes

In [153]:
si_cannes = si[si.Ville == "CANNES"]
si_cannes

Siren   Nic  \
Dénomination                      
RIEN 100 RIEN   880259460  17.0   
LM DELIVERY     884390501  10.0   
DRINK & TEST    882542665  14.0   
SAS PRIME AZUR  882559487  13.0   
CONFIDENTIAL    887472850  15.0   
...                   ...   ...   
BOX'IN          881428072  14.0   
KONSEILS        881023493  11.0   
VOLGA           881007900  15.0   
BC LINE         881236541  10.0   
1984 EVENT      890672595  18.0   

                                                  Forme Juridique Code APE  \
Dénomination                                                                 
RIEN 100 RIEN     Societe par actions simplifiee a associe unique      NaN   
LM DELIVERY                        Societe par actions simplifiee      NaN   
DRINK & TEST      Societe par actions simplifiee a associe unique    9609Z   
SAS PRIME AZUR    Societe par actions simplifiee a associe unique    9609Z   
CONFIDENTIAL                       Societe par actions simplifiee      NaN   
...                                                           ...      ...   
BOX'IN            Societe par actions simplifiee a associe unique      NaN   
KONSEILS          Societe par actions simplifiee a associe unique    4690Z   
VOLGA           Societe a responsabilite limitee a associe unique    6810Z   
BC LINE                            Societe par actions simplifiee    4511Z   
1984 EVENT      Societe a responsabilite limitee a associe unique    8230Z   

                                               Secteur d'activité  \
Dénomination                                                        
RIEN 100 RIEN                                                 NaN   
LM DELIVERY                                                   NaN   
DRINK & TEST                    Autres services personnels n.c.a.   
SAS PRIME AZUR                  Autres services personnels n.c.a.   
CONFIDENTIAL                                                  NaN   
...                                                           ...   
BOX'IN                                                        NaN   
KONSEILS        Commerce de gros (commerce interentreprises) n...   
VOLGA                Activites des marchands de biens immobiliers   
BC LINE         Commerce de voitures et de vehicules automobil...   
1984 EVENT      Organisation de foires, salons professionnels ...   

                                      Adresse  Code postal   Ville  \
Dénomination                                                         
RIEN 100 RIEN          31 BOULEVARD DU MOULIN         6400  CANNES   
LM DELIVERY     97 BOULEVARD DE LA REPUBLIQUE         6400  CANNES   
DRINK & TEST                       9 RUE PONS         6400  CANNES   
SAS PRIME AZUR              18 RUE LORD BYRON         6400  CANNES   
CONFIDENTIAL      6 RUE ALPHONSE DE LAMARTINE         6150  CANNES   
...                                       ...          ...     ...   
BOX'IN                       4 AVENUE DU NORD         6150  CANNES   
KONSEILS                 12 AVENUE DES ARLUCS         6150  CANNES   
VOLGA                     68 BOULEVARD CARNOT         6400  CANNES   
BC LINE                      37 RUE D ANTIBES         6400  CANNES   
1984 EVENT          6 AVENUE JEAN DE NOAILLES         6400  CANNES   

                Num. dept.      Département  ... Nom commercial  \
Dénomination                                 ...                  
RIEN 100 RIEN            6  Alpes-Maritimes  ...            NaN   
LM DELIVERY              6  Alpes-Maritimes  ...            NaN   
DRINK & TEST             6  Alpes-Maritimes  ...            NaN   
SAS PRIME AZUR           6  Alpes-Maritimes  ...            NaN   
CONFIDENTIAL             6  Alpes-Maritimes  ...            NaN   
...                    ...              ...  ...            ...   
BOX'IN                   6  Alpes-Maritimes  ...            NaN   
KONSEILS                 6  Alpes-Maritimes  ...            NaN   
VOLGA                    6  Alpes-Maritimes  ...            NaN   
BC 

## Sélection des 7 colonnes qui nous interesse (parmis les 28 présentes dans le jeu de données initial)

In [157]:
for idx,column in enumerate(si_cannes.columns):
    print(idx,column)

0 Siren
1 Nic
2 Forme Juridique
3 Code APE
4 Secteur d'activité
5 Adresse
6 Code postal
7 Ville
8 Num. dept.
9 Département
10 Région
11 Code Greffe
12 Greffe
13 Date immatriculation
14 Date radiation
15 Statut
16 Geolocalisation
17 Date de publication
18 Nom commercial
19 Date immatriculation origine
20 Sigle
21 Devise
22 Durée
23 Date cloture 1er exercice
24 Date arreté des comptes
25 Etat
26 etat_pub
27 fiche_identite


In [158]:
si_cannes.iloc[1,2,6,7,8,9,18]
si_cannes


 mycolumns = ['GrLivArea','SalePrice'] 
>>> df[mycolumns]

IndexingError: Too many indexers

In [154]:
print("Le jeu de données recense", si_cannes.shape[0], "entreprises avec", si_cannes.shape[1], "variables")

Le jeu de données recense 368 entreprises avec 28 variables


### Regroupement des entreprises dans un meme dataframe

Cette variable regroupe l'ensemble des données relatives aux entreprises des cinq villes de l'agglomération.

In [55]:
cinq_villes = si[(si.Ville == "CANNES") | (si.Ville == "LE CANNET") | (si.Ville == "THEOULE SUR MER") | (si.Ville == "MOUGINS") | (si.Ville == "MANDELIEU LA NAPOULE")]
cinq_villes

Siren   Nic  \
Dénomination                                        
SASU SEBTAXI                      880244686  15.0   
RIEN 100 RIEN                     880259460  17.0   
LM DELIVERY                       884390501  10.0   
CUISINE ET BAIN 06                882487705  15.0   
DRINK & TEST                      882542665  14.0   
...                                     ...   ...   
L.A DISTRIBUTION & LOGISTIQUE     881254692  18.0   
SERVEURS AND CO                   881237275  14.0   
DHOLE.NET                         881188353  18.0   
1984 EVENT                        890672595  18.0   
KODU YACHT AND PROPERTY SERVICES  890714173  14.0   

                                                                    Forme Juridique  \
Dénomination                                                                          
SASU SEBTAXI                        Societe par actions simplifiee a associe unique   
RIEN 100 RIEN                       Societe par actions simplifiee a associe unique   
LM DELIVERY                                          Societe par actions simplifiee   
CUISINE ET BAIN 06                Societe a responsabilite limitee a associe unique   
DRINK & TEST                        Societe par actions simplifiee a associe unique   
...                                                                             ...   
L.A DISTRIBUTION & LOGISTIQUE     Societe a responsabilite limitee a associe unique   
SERVEURS AND CO                     Societe par actions simplifiee a associe unique   
DHOLE.NET                                          Societe a responsabilite limitee   
1984 EVENT                        Societe a responsabilite limitee a associe unique   
KODU YACHT AND PROPERTY SERVICES                     Societe par actions simplifiee   

                                 Code APE  \
Dénomination                                
SASU SEBTAXI                          NaN   
RIEN 100 RIEN                         NaN   
LM DELIVERY                           NaN   
CUISINE ET BAIN 06                  4334Z   
DRINK & TEST                        9609Z   
...                                   ...   
L.A DISTRIBUTION & LOGISTIQUE       4531Z   
SERVEURS AND CO                     6201Z   
DHOLE.NET                           4791A   
1984 EVENT                          8230Z   
KODU YACHT AND PROPERTY SERVICES    8211Z   

                                                                 Secteur d'activité  \
Dénomination                                                                          
SASU SEBTAXI                                                                    NaN   
RIEN 100 RIEN                                                                   NaN   
LM DELIVERY                                                                     NaN   
CUISINE ET BAIN 06                                  Travaux de peinture et vitrerie   
DRINK & TEST                                      Autres services personnels n.c.a.   
...                                                                             ...   
L.A DISTRIBUTION & LOGISTIQUE            Commerce de gros d'equipements automobiles   
SERVEURS AND CO                                          Programmation informatique   
DHOLE.NET                                    Vente a distance sur catalogue general   
1984 EVENT                        Organisation de foires, salons professionnels ...   
KODU YACHT AND PROPERTY SERVICES         Services administratifs combines de bureau   

                                                        Adresse  Code postal  \
Dénomination                                                                   
SASU SEBTAXI                            8 TRAVERSE DE L ESTELLE         6110   
RIEN 100 RIEN                            31 BOULEVARD DU MOULIN         6400   
LM DELIVERY                       97 BOULEVARD DE LA REPUBLIQUE         6400   
CUISINE ET BAIN 06                      101 CHEMIN SAINT JOSEPH         6110   
DRINK & TEST      

### Dimension du jeu de données

In [151]:
print("Le jeu de données recense", cinq_villes.shape[0], "entreprises avec", cinq_villes.shape[1], "variables")

Le jeu de données recense 704 entreprises avec 28 variables


### Valeurs manquantes

Le jeu de données est pour les colonnes qui nous interesse partiellement complet.

* Dénomination
* Siren
* Secteur d'activité
* Adresse
* Code postal
* Ville
* Géolocalisation

On remarque que les colonnes "Secteur d'activité", "Adresse", "Géolocalisation" ont quelques valeurs manquantes.

In [56]:
cinq_villes.isna().sum()

Siren                             0
Nic                              25
Forme Juridique                   0
Code APE                        193
Secteur d'activité              193
Adresse                           0
Code postal                       0
Ville                             0
Num. dept.                        0
Département                       0
Région                            0
Code Greffe                       0
Greffe                            0
Date immatriculation              0
Date radiation                  704
Statut                            0
Geolocalisation                 126
Date de publication             189
Nom commercial                  704
Date immatriculation origine    190
Sigle                           497
Devise                          189
Durée                           704
Date cloture 1er exercice       366
Date arreté des comptes           1
Etat                            189
etat_pub                         26
fiche_identite              

### Décompte des doublons

In [63]:
cinq_villes.duplicated().sum()

0

In [64]:
cinq_villes.isna().sum()

Siren                             0
Nic                              25
Forme Juridique                   0
Code APE                        193
Secteur d'activité              193
Adresse                           0
Code postal                       0
Ville                             0
Num. dept.                        0
Département                       0
Région                            0
Code Greffe                       0
Greffe                            0
Date immatriculation              0
Date radiation                  704
Statut                            0
Geolocalisation                 126
Date de publication             189
Nom commercial                  704
Date immatriculation origine    190
Sigle                           497
Devise                          189
Durée                           704
Date cloture 1er exercice       366
Date arreté des comptes           1
Etat                            189
etat_pub                         26
fiche_identite              

### Notes générales

On remarque que la plupart des entreprises de l'agglomération appartiennent aux formes juridiques suivantes:
    
* Societe par actions simplifiee a associe unique                                           
* Societe par actions simplifiee                                                            
* Societe a responsabilite limitee                                                          
* Societe a responsabilite limitee a associe unique 

Cela pourra nous etre utile pour l'analyse de notre étude socio-démographique vis-à-vis de la production de ces entreprises en matières de déchets.

In [224]:
forme_juridique = cinq_villes['Forme Juridique'].value_counts()
forme_juridique

Societe par actions simplifiee a associe unique                         261
Societe par actions simplifiee                                          203
Societe a responsabilite limitee                                        121
Societe a responsabilite limitee a associe unique                       105
Societe en nom collectif                                                 12
Societe par actions simplifiee  a associe unique et capital variable      2
Name: Forme Juridique, dtype: int64

In [225]:
fj = cinq_villes.groupby(['Forme Juridique']).size()
fj

Forme Juridique
Societe a responsabilite limitee                                        121
Societe a responsabilite limitee a associe unique                       105
Societe en nom collectif                                                 12
Societe par actions simplifiee                                          203
Societe par actions simplifiee  a associe unique et capital variable      2
Societe par actions simplifiee a associe unique                         261
dtype: int64

In [182]:
#cinq_villes.loc[cinq_villes["Dénomination"] == 'RIEN 100 RIEN', :]

Dénomination      Siren   Nic  \
68  RIEN 100 RIEN  880259460  17.0   

                                    Forme Juridique Code APE  \
68  Societe par actions simplifiee a associe unique      NaN   

   Secteur d'activité                 Adresse  Code postal   Ville  \
68                NaN  31 BOULEVARD DU MOULIN         6400  CANNES   

    Num. dept.  ... Nom commercial Date immatriculation origine  Sigle Devise  \
68           6  ...            NaN                   2020-01-03    NaN  Euros   

   Durée  Date cloture 1er exercice Date arreté des comptes     Etat etat_pub  \
68   NaN                 2020-12-31                   31/12  Nouveau        A   

                                       fiche_identite  
68  https://www.infogreffe.fr/infogreffe/ficheIden...  

[1 rows x 29 columns]

In [57]:
secteur_activite = cinq_villes["Secteur d'activité"].value_counts()
secteur_activite

EN INSTANCE DE CHIFFREMENT                                                       49
Activites des marchands de biens immobiliers                                     35
Conseil pour les affaires et autres conseils de gestion                          18
Agences immobilieres                                                             16
Restauration de type rapide                                                      16
                                                                                 ..
Collecte et traitement des eaux usees                                             1
Comm. de gros (comm. interent.) de composants et d'equi. electro. et telecom.     1
Autres activites liees au sport                                                   1
Transports routiers de fret de proximite                                          1
Reparation et maintenance navale                                                  1
Name: Secteur d'activité, Length: 141, dtype: int64

In [172]:
nbre_entreprise_ville=[sum(i) for i in cinq_villes]
nbre_entreprise_ville

TypeError: unsupported operand type(s) for +: 'int' and 'str'

In [ ]:
donneeslabel = donnees[donnees['Label'].notna()][['TCLabs MetaCode', 'Label']]
show_Label = pd.DataFrame(columns = ['Label','TC MetaCode'])

In [67]:
from sqlalchemy import create_engine

engine = create_engine('mysql+pymysql://simplon:mdp@@localhost:3306/stage', echo=True)

In [69]:
entreprises_par_sa.rename(columns={'entreprises_par_sa' : 'sa_nbre_établissement'}).to_sql('country', con=engine, index = False, if_exists = 'append')

2020-11-22 23:55:45,535 INFO sqlalchemy.engine.base.Engine DESCRIBE `country`
2020-11-22 23:55:45,537 INFO sqlalchemy.engine.base.Engine {}
2020-11-22 23:55:45,539 INFO sqlalchemy.engine.base.Engine ROLLBACK
2020-11-22 23:55:45,545 INFO sqlalchemy.engine.base.Engine 
CREATE TABLE country (
	`Catégorie` TEXT, 
	`France` BIGINT, 
	`Mougins` BIGINT, 
	`Mandelieu-la-Napoule` BIGINT, 
	`Le Cannet` BIGINT, 
	`Théoule-sur-Mer` BIGINT, 
	`Cannes` BIGINT, 
	`France.1` FLOAT(53), 
	`Mougins.1` FLOAT(53), 
	`Mandelieu-la-Napoule.1` FLOAT(53), 
	`Le Cannet.1` FLOAT(53), 
	`Théoule-sur-Mer.1` FLOAT(53)
)


2020-11-22 23:55:45,545 INFO sqlalchemy.engine.base.Engine {}
2020-11-22 23:55:45,966 INFO sqlalchemy.engine.base.Engine COMMIT
2020-11-22 23:55:45,985 INFO sqlalchemy.engine.base.Engine BEGIN (implicit)
2020-11-22 23:55:45,987 INFO sqlalchemy.engine.base.Engine INSERT INTO country (`Catégorie`, `France`, `Mougins`, `Mandelieu-la-Napoule`, `Le Cannet`, `Théoule-sur-Mer`, `Cannes`, `France.1`, `Mo